src/map/script.c

```diff
@@ -13573,14 +13573,28 @@ BUILDIN(atcommand) {
 
 /*==========================================
  * Displays a message for the player only (like system messages like "you got an apple" )
+ * usage : dispbottom "<message>"{,<color>{,<account ID>}};
  *------------------------------------------*/
-BUILDIN(dispbottom)
-{
-	TBL_PC *sd=script->rid2sd(st);
+BUILDIN(dispbottom) {
+	TBL_PC *sd = NULL;
+	int color = 0;
 	const char *message;
-	message=script_getstr(st,2);
-	if(sd)
-		clif_disp_onlyself(sd,message,(int)strlen(message));
+
+	message = script_getstr(st,2);
+	if (script_hasdata(st,3))
+		color = script_getnum(st,3);
+	if (script_hasdata(st,4))
+		sd = map->id2sd(script_getnum(st,4));
+	else
+		sd = script->rid2sd(st);
+
+	if (sd) {
+		if (color > 0)
```

Comment:
This condition is incorrect. It can't handle the case where you want the message to be printed in black (`0x000000`)